From jesper.juhl@gmail.com Fri Mar 21 14:56:00 2008
From: Jesper Juhl <jesper.juhl@gmail.com>
Date: Fri, 21 Mar 2008 22:55:45 +0100 (CET)
Subject: USB: test for NULL return from platform_get_resource() in ohci_hcd_sm501_drv_remove()
Cc: Roman Weissgaerber <weissg@vienna.at>,  Lennert Buytenhek <kernel@wantstofly.org>, Greg KH <greg@kroah.com>
Message-ID: <alpine.LNX.1.00.0803212250210.9597@dragon.funnycrock.com>



platform_get_resource() may return null, so although it seems it will never
do so here unless there's a bug elsewhere, it does no harm to be defensive
and test.

Signed-off-by: Jesper Juhl <jesper.juhl@gmail.com>
Acked-by: David Brownell <dbrownell@users.sourceforge.net>
Acked-by: Magnus Damm <magnus.damm@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/ohci-sm501.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/usb/host/ohci-sm501.c
+++ b/drivers/usb/host/ohci-sm501.c
@@ -199,7 +199,8 @@ static int ohci_hcd_sm501_drv_remove(str
 	usb_put_hcd(hcd);
 	dma_release_declared_memory(&pdev->dev);
 	mem = platform_get_resource(pdev, IORESOURCE_MEM, 1);
-	release_mem_region(mem->start, mem->end - mem->start + 1);
+	if (mem)
+		release_mem_region(mem->start, mem->end - mem->start + 1);
 
 	/* mask interrupts and disable power */
 
